### DISCUSSION TITLE
[alerting unified view] alerts can be filter with either externalLabels or instance labels but not both

### DISCUSSION THREAD

**Discussion by AbelThorne** [grafana/grafana#49210] [Alerting]
**What happened**:
On the Unified Alerting View, when using labels to filter alerts, if only using external labels (associated with the rule not the alert instance), it is fine. Same things if using only alert instance labels. However, when using labels from both family, the alert list is always empty, regardless of the labels value. 

**What you expected to happen**:
Expected behaviour is to be able to filter alerts using both instance labels and external labels as is possible when doing queries on the ALERTS and ALERTS_FOR_STATE metrics. 

**How to reproduce it (as minimally and precisely as possible)**:
On the unified alerting view, select a datasource with external labels (such as Thanos for instance), filter alerts using the default external label "prometheus" and any other instance label. List of alerts will be empty. 

**Anything else we need to know?**:
As of now, there is more flexibility to display specific alerts using a Table panel and ALERTS/ALERTS_FOR_STATE metrics but the look and feel is much worse than when using the Alert List Panel or the Unified Alerting View. Maybe it would be interesting to allow full queries in the Alert List Panel to choose with precision the kind of alerts to display but with the current UI (colored labels and collapsible description) that is really great. 

**Environment**:
- Grafana version: 8.5.0
- Data source type & version: Prometheus Operator 0.56.2 & Thanos 0.24 
- OS Grafana is installed on: Helm Chart on Kubernetes 
- User OS & Browser: MacOS
- Grafana plugins:
- Others:

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
